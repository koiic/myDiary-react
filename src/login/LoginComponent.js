import React, { Component, Fragment } from 'react';

// third party library
import { Redirect } from 'react-router-dom';
import Loader from 'react-loader';

/**
 * @param {object} props
 * @desc renders login page
 */
class LoginComponent extends Component {
  state = {
    username: '',
    password: '',
  };

  handleChange = (event) => {
    event.preventDefault();
    this.setState({ [event.target.name]: event.target.value });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const { username, password } = this.state;
    this.props.login({ username, password });
  };

  render() {
    const { username, password } = this.state;

    const { props } = this;
    const loading = props.auth.login.processing
      ? { display: 'block' }
      : { display: 'none' };
    if (props.auth.isAuth === true) {
      return <Redirect to='/' />;
    }

    return (
      <Fragment>
        <div className='signup'>
          <div className='header'>
            <div className='navbar fixed-top'>
              <div className='shelter'>
                <div className='navbar__nav nav__logo nav__underline'>
                  <i className='fa fa-book' />
                  <span className='highlight'>My</span>-Diarry
                </div>
                <div className='navbar__nav navbar__nav-toggle'>
                  <i className='fa fa-bars' />
                </div>
                <nav className='nav__items nav__items-right'>
                  <div className='navbar__nav pr-0'>
                    <a href='signup.html'>
                      <button className='btn btn-light'>Create Account</button>
                    </a>
                  </div>
                </nav>
              </div>
            </div>
          </div>
          <section>
            <div className='shelter'>
              <div className='row'>
                <div className='column' />
                <div className='column'>
                  <div className='form'>
                    <h1 className='text-center nav__logo app-color'>
                      Welcome Back
                    </h1>
                    <div className='form-card'>
                      <div style={loading}>
                        <Loader color='#0FC86F' speed={1} className='spinner' />
                      </div>
                      <form onSubmit={this.handleSubmit}>
                        <div className='form-row'>
                          <label htmlFor='name'>Username:</label>
                          <input
                            type='text'
                            className='form-control'
                            name='username'
                            value={username}
                            onChange={this.handleChange}
                            placeholder='Enter your username'
                            required
                            minLength={3}
                            maxLength={30}
                          />
                        </div>
                        <div className='form-row'>
                          <label htmlFor='name'>Password</label>
                          <input
                            type='password'
                            className='form-control'
                            required
                            placeholder='Enter your password'
                            name='password'
                            value={password}
                            onChange={this.handleChange}
                          />
                        </div>
                        <button className='btn account__btn btn-block'>
                          Login
                        </button>
                      </form>
                    </div>
                    <p>
                      Yet to Sign Up?Sign Up <a href='signup.html'>here</a>
                    </p>
                  </div>
                </div>
                <div className='column' />
              </div>
            </div>
          </section>
          <section>
            <div className='footer'>
              <div className='navbar py-1 bg-white'>
                <div className='shelter'>
                  <nav className='nav__items nav__items-right'>
                    <div className='navbar__nav'>
                      <div className='display'>
                        <a href='#' className='dodgerblue'>
                          Â© Ismail Ibrahim[Andela Cycle34 2018].
                        </a>
                      </div>
                    </div>
                  </nav>
                </div>
              </div>
            </div>
          </section>
        </div>
      </Fragment>
    );
  }
}

export default LoginComponent;
