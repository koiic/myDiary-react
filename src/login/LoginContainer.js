import { connect } from 'react-redux';

// components
import LoginComponent from './LoginComponent';

// action
import { userLogin } from './loginAction';

const mapDispatchToProps = dispatch => ({
  login: detail => dispatch(userLogin(detail))
});
const mapStateToProps = state => ({
  auth: state.authReducer
});

export default connect(
  mapStateToProps, mapDispatchToProps
)(LoginComponent);
