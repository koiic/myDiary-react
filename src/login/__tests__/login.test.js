// React Libraries
import React from 'react';

// third-party libraries
import { shallow, configure } from 'enzyme';

import { Provider } from 'react-redux';

// modules imports
import configureStore from '../../app/mainStore/store';

// modules importation
import LoginContainer from '../LoginContainer';
import LoginComponent from '../LoginComponent';

// SnapShot Test
describe('Renders the singup page', () => {
  it('should render the signup form', () => {
    const store = configureStore();
    const wrapper = shallow(
      <Provider store={store}>
        <LoginContainer />
      </Provider>
    );
    expect(wrapper.exists()).toBe(true);
  });
});

describe('Renders the singup page', () => {
  it('should render the signup form', () => {
    const component = shallow(
      <LoginComponent auth={{ login: { processing: true } }} />
    );
    expect(component).toMatchSnapshot();
  });
});
