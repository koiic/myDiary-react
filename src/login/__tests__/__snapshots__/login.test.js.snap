// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the singup page should render the signup form 1`] = `
<Fragment>
  <div
    className="signup"
  >
    <div
      className="header"
    >
      <div
        className="navbar fixed-top"
      >
        <div
          className="shelter"
        >
          <div
            className="navbar__nav nav__logo nav__underline"
          >
            <i
              className="fa fa-book"
            />
            <span
              className="highlight"
            >
              My
            </span>
            -Diarry
          </div>
          <div
            className="navbar__nav navbar__nav-toggle"
          >
            <i
              className="fa fa-bars"
            />
          </div>
          <nav
            className="nav__items nav__items-right"
          >
            <div
              className="navbar__nav pr-0"
            >
              <a
                href="signup.html"
              >
                <button
                  className="btn btn-light"
                >
                  Create Account
                </button>
              </a>
            </div>
          </nav>
        </div>
      </div>
    </div>
    <section>
      <div
        className="shelter"
      >
        <div
          className="row"
        >
          <div
            className="column"
          />
          <div
            className="column"
          >
            <div
              className="form"
            >
              <h1
                className="text-center nav__logo app-color"
              >
                Welcome Back
              </h1>
              <div
                className="form-card"
              >
                <div
                  style={
                    Object {
                      "display": "block",
                    }
                  }
                >
                  <Component
                    className="spinner"
                    color="#0FC86F"
                    component="div"
                    loadedClassName="loadedContent"
                    parentClassName="loader"
                    speed={1}
                  />
                </div>
                <form
                  onSubmit={[Function]}
                >
                  <div
                    className="form-row"
                  >
                    <label
                      htmlFor="name"
                    >
                      Username:
                    </label>
                    <input
                      className="form-control"
                      maxLength={30}
                      minLength={3}
                      name="username"
                      onChange={[Function]}
                      placeholder="Enter your username"
                      required={true}
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    className="form-row"
                  >
                    <label
                      htmlFor="name"
                    >
                      Password
                    </label>
                    <input
                      className="form-control"
                      name="password"
                      onChange={[Function]}
                      placeholder="Enter your password"
                      required={true}
                      type="password"
                      value=""
                    />
                  </div>
                  <button
                    className="btn account__btn btn-block"
                  >
                    Login
                  </button>
                </form>
              </div>
              <p>
                Yet to Sign Up?Sign Up 
                <a
                  href="signup.html"
                >
                  here
                </a>
              </p>
            </div>
          </div>
          <div
            className="column"
          />
        </div>
      </div>
    </section>
    <section>
      <div
        className="footer"
      >
        <div
          className="navbar py-1 bg-white"
        >
          <div
            className="shelter"
          >
            <nav
              className="nav__items nav__items-right"
            >
              <div
                className="navbar__nav"
              >
                <div
                  className="display"
                >
                  <a
                    className="dodgerblue"
                    href="#"
                  >
                    Â© Ismail Ibrahim[Andela Cycle34 2018].
                  </a>
                </div>
              </div>
            </nav>
          </div>
        </div>
      </div>
    </section>
  </div>
</Fragment>
`;
